PythonVisualizations/SkipList.py

```diff
@@ -563,21 +563,83 @@ def blink(self, n, color="blue"):
         ########################
         
     def makeButtons(self):
-        vcmd = (self.window.register(numericValidate),\
-                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')        
-        self.addOperation("insert", lambda: self.click(self.insert), numArguments=1, validationCmd=vcmd)
-        self.addOperation("delete", lambda: self.click(self.delete), numArguments=1, validationCmd=vcmd)
-        self.addOperation("fill", lambda: self.click(self.fill), numArguments=1, validationCmd=vcmd)
-        self.addOperation("search", lambda: self.click(self.search), numArguments=1, validationCmd=vcmd)
-        self.addAnimationButtons()        
+        vcmd = (self.window.register(numericValidate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+        searchButton = self.addOperation(
+            "Search", lambda: self.clickSearch(), numArguments=1,
+            validationCmd=vcmd, helpText="Click to enter number")
+        insertButton = self.addOperation(
+            "Insert", lambda: self.clickInsert(), numArguments=1,
+            validationCmd=vcmd)        
+        deleteButton = self.addOperation(
+            "Delete", lambda: self.clickDelete(), numArguments=1,
+            validationCmd=vcmd)
+        fillButton = self.addOperation(
+            "Fill", lambda: self.clickFill(), numArguments=1, 
+            validationCmd=vcmd)
+        # this makes the pause, play and stop buttons 
+        self.addAnimationButtons()
+        return [searchButton, insertButton, deleteButton, fillButton]
     
-    # already an onClick()    
-    def click(self, call):
-        val = int(self.getArgument())
-        self.setArgument()
-        self.startAnimations()
-        call(val)
-        self.stopAnimations()

```

Comment:
Replacing the use of the `click()` method with calls to the `clickSearch()`,  `clickInsert()`, and so on methods has disabled the animations.  That's because the call to `startAnimations()` is no longer happening.
This could be fixed by two methods:
 1. restoring the `click()` method but without the getting the user argument into `val` and just invoking `call()`; and using callbacks like `lambda: self.click(self.clickFill)`
 2. converting the operations like `insert()` and `fill()` to use the call stack API and call `callEnviron = self.createCallEnvironment()`, `self.startAnimations()`, and `self.cleanUp(callEnviron)` at the beginning and end of the operations.

Option 2 is probably the best because it gets the module ready for showing and highlighting code as the animation runs.